### ISSUE TITLE
Test failed on Arch Linux against Python 3.12

### ISSUE THREAD

**Issue by cybaol** [gaogaotiantian/viztracer#430]
Error Log: [test-error.log](https://github.com/gaogaotiantian/viztracer/files/15163635/test-error.log)

Note: `python-orjson`, `python-rich` are pre-installed on my machine.

**Comment by cybaol** [comment]
Sorry so late for reply.
Now, I have passed all tests based on the minor changes I've made.
Following is my change:
```shell
  # Avoid exceeding maximum recursion depth.
  sed -i "s|call_self(1000)|call_self(888)|g" tests/test_vcompressor.py

  # Disable external processor testing because we don't have it.
  sed -i "s|use_external_processor=True|use_external_processor=False|g" tests/test_viewer.py

  # Attach is not working, this is some black magic and is OS-dependent.
  rm tests/test_remote.py
```

For the first one, on my machine, the maximum recursion depth is exceeded when exceeding 996.

**Comment by gaogaotiantian** [comment]
Free free to pick out the tests that do not suit you.

For the tests you found issues:
1. That recursion test is not well written. The direct trigger is that you are using pytest which will introduce a few extra frames. That should be avoided and I'll probably fix that.
2. The external processor is something that the script should download from google. It only enables the external processor so if you don't need that, you can skip the test.
3. Attaching is some black magic and replies on VM. Arch Linux is not officially supported so yes it might not work.

Anyway, I probably won't change the tests except for the first one, but you can just do what you need to package it.

**Comment by cybaol** [comment]
I noticed that there are still two test errors about `orjson`(latest version from pip).
Here is [build.log](https://github.com/gaogaotiantian/viztracer/files/15376867/build.log).
Previous test is using `json` from python standard library.

**Comment by gaogaotiantian** [comment]
I can take a look this weekend probably.

**Comment by cybaol** [comment]
好

**Comment by gaogaotiantian** [comment]
This is due to some extreme tests for vcompressor. `orjson` is more strict to the input data. I fixed those tests in #447 so the tests should work with `orjson` now,
